src/canmatrix/canmatrix.py

```diff
@@ -1431,6 +1511,19 @@ def frame_by_id(self, arbitration_id):  # type: (ArbitrationId) -> typing.Union[
                 return test
         return None
 
+    def frame_by_pgn(self, pgn):  # type: (int) -> typing.Union[Frame, None]
+        """Get Frame by pgn (j1939).
+
+        :param int pgn: pgn to search for
+        :rtype: Frame or None
+        """
+
+        for test in self.frames:
+            if test.arbitration_id.pgn == canmatrix.ArbitrationId.from_pgn(pgn).pgn:
```

Comment:
```suggestion
            if test.arbitration_id.pgn == pgn:
```

Reply:
`canmatrix.ArbitrationId.from_pgn(pgn).pgn` instead of just `pgn` is needed to do the `pf >= 240` check and rewrite the pgn